test/tests/functional/pbs_calendaring.py

```diff
@@ -137,3 +137,45 @@ def test_topjob_start_time_of_subjob(self):
         self.assertIn('estimated.start_time', job_arr[0])
         errmsg = jid + ";Error in calculation of start time of top job"
         self.scheduler.log_match(errmsg, existence=False, max_attempts=10)
+
+    def test_topjob_fail(self):
+        """
+        Test that when we fail to add a job to the calendar it doesn't
+        take up a topjob slot.  The server's backfill_depth is 1 by default,
+        so we just need to submit a job that can never run and a job that can.
+        The can never run job will fail to be added to the calendar and the
+        second job will be.
+        """
+        attrs = {'resources_available.ncpus': 1}
+        self.server.create_vnodes('vn', attrib=attrs, num=2,
```

Comment:
just curious, any particular reason why you decided to create 2 vnodes instead of just using the one available and setting its ncpus?

Reply:
I needed to create a situation where a job could never run, but the scheduler doesn't realize that.  In this case there are two nodes with 1 cpu on it.  The can never run job is requesting 1 node with 2 cpus.  We have to be careful when creating can never run jobs, because the scheduler is really good at detecting that jobs can never run.  If it does, it won't add the job to the calendar.  If I turned it around and had one node with 2 cpus and requested 2 nodes with 1 cpu, the scheduler would have caught it as a can never run job and not added it to the calendar

Reply:
Thanks for clarifying. Can you please add a comment explaining this in the test? It's not really obvious from the code.

Reply:
I added a comment.  Take a look and see if you like it.